### PR TITLE
Test that History is associated with Document, not Window

### PR DESCRIPTION
**Pull request by domenic**
This is done by expanding the object-association.js helper and using it.

See https://github.com/whatwg/html/issues/2566.

### CODE REVIEW

**Comment by domenic** [comment]
I've tested Firefox (per-Window) and Chrome (per-Document). Given that they disagree, and this is simple enough to change, we can let Safari be the tiebreaker when the results come in...

**Comment by domenic** [comment]
Safari seems to match Chrome and associate History with Document, not Window, so we'll leave the spec and this test as-is. This is thus ready for review.

File: common/object-association.js [hunk 1]
```diff
@@ -64,5 +78,5 @@ window.testIsPerWindow = propertyName => {
 
     iframe.src = "/common/blank.html";
     document.body.appendChild(iframe);
-  }, `document.open() must replace window.${propertyName}`);
-};
+  }, `document.open() must not replace window.${propertyName}`);
```
Comment (domfarolino) [inline common/object-association.js:81]: Nice catch

**Review by domfarolino — APPROVED** [review]

File: common/object-association.js [hunk 2]
```diff
@@ -38,14 +36,30 @@ window.testIsPerWindow = propertyName => {
...
 
-  // Per spec, document.open() should not change any of the Window state.
+  // Per spec, discarding a browsing context should not change any of the global objects.
+  test(() => {
+    const iframe = document.createElement("iframe");
+    document.body.appendChild(iframe);
+    const frame = iframe.contentWindow;
+
+    const before = frame[propertyName];
+    assert_true(before !== undefined && before !== null, `window.${propertyName} must be implemented`);
```
[the remark this answers is not in the document]
Comment (domenic) [reply common/object-association.js:53]: Yeah, good idea, I think that did not exist when I first wrote that helper.

**Pull request merged**